src/js/grapl-cdk/lib/engagement.ts

```diff
@@ -222,7 +223,12 @@ export class EngagementEdge extends cdk.NestedStack {
 }
 
 export class EngagementNotebook extends cdk.NestedStack {
-    constructor(scope: cdk.Construct, id: string, props: GraplServiceProps) {
+    constructor(
```

Comment:
Build is currently broken, call to this in lib/grapl-cdk-stack.ts doesn't pass the new arg

Reply:
Instead of adding params to constructor, I'd started extending the props argument, which seems to be std practice for CDK. I think we should do that here.

For example:
```
export interface NodeIdentifierProps extends GraplServiceProps {
    writesTo: s3.IBucket;
}

...

    constructor(parent: cdk.Construct, id: string, props: NodeIdentifierProps) {
```

https://github.com/grapl-security/grapl/blob/staging/src/js/grapl-cdk/lib/grapl-cdk-stack.ts#L71